src/i2c.rs

```diff
@@ -617,6 +671,28 @@ impl<PINS> I2c<I2C3, PINS> {
     }
 }
 
+#[cfg(any(feature = "stm32f413", feature = "stm32f423",))]
+impl<PINS> FMPI2c<FMPI2C, PINS> {
+    pub fn fmpi2c(i2c: FMPI2C, pins: PINS, speed: KiloHertz, clocks: Clocks) -> Self
+    where
+        PINS: Pins<FMPI2C>,
+    {
+        // NOTE(unsafe) This executes only during initialisation
+        let rcc = unsafe { &(*RCC::ptr()) };
+
+        // Enable clock for FMPI2C
+        rcc.apb1enr.modify(|_, w| w.i2c4en().set_bit());
```

Comment:
Isn't it racy?

Reply:
Yes it is, any suggestions?

Reply:
Critical section, bit banging or &mut Rcc as parameter?

Reply:
Yeah, the usual suspects. ;)

At least the race is consistent with the already existing other I2C impls. I guess we might want to fix them all in the same way in one swoop?

Reply:
As you wish, there is a lot of racy code in this crate. Maybe we should open issues to track them?

Reply:
Let's do that.

Reply:
I don't have the time right now to check all the places which require attention but I've opened #171 to keep track of required fixes.